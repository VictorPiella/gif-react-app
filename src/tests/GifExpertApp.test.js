import React from 'react';
import '@testing-library/jest-dom'

import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp'


describe('Test <GifExpertApp />', () => {
  
  test('show the component correctly', () => {
    
    const wrapper = shallow( <GifExpertApp  /> );
    expect ( wrapper ).toMatchSnapshot();
  
  });
  test('should show list of categories', () => {

    const categories = ['One Pice', 'Suits'];
    const wrapper = shallow( <GifExpertApp  defaultCategories={ categories }/> );
    expect ( wrapper ).toMatchSnapshot();
    expect ( wrapper.find('GifGrid').length ).toBe( categories.length );
    
  })
  
  
})
