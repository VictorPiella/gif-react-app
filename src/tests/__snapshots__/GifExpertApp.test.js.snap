// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifExpertApp /> should show list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Pice"
      key="One Pice"
    />
    <GifGrid
      category="Suits"
      key="Suits"
    />
  </ol>
</Fragment>
`;

exports[`Test <GifExpertApp /> show the component correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
